src/components/TabButtons.tsx

```diff
@@ -0,0 +1,45 @@
+import Link from "next/link";
+import { usePathname, useSelectedLayoutSegment } from "next/navigation";
+
+interface TabButtonsProps {
+  queries: string[];
+}
+
+const TabButtons = ({ queries }: TabButtonsProps) => {
+  const pathname = usePathname() ?? "";
+  const segment = useSelectedLayoutSegment();
+
+  return (
+    <div style={{ width: "fit-content" }}>
+      <div className="flex gap-6">
+        {queries.map((query, index) => {
+          const isTabSelected =
+            segment === query || (segment === null && index === 0);
+          const href =
```

Comment:
Can you guys explain this logic? It seems like `pathname + "/" + query` and `pathname.split("/")...` does the same thing, or am I missing something?

Reply:
The first one (pathname + "/" + query) adds to the end of the pathname (i.e. clicking on one of the tabs for the first time from /home) and the other one (.split...) basically replaces the last section of the path and replaces it with the new query (i.e. switching between tabs case)

Reply:
Got it! Thanks for clarifying